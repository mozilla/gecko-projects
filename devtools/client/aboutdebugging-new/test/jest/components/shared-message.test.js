/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the shared/Message component.
 */

const renderer = require("react-test-renderer");
const React = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");

const { MESSAGE_LEVEL } = require("devtools/client/aboutdebugging-new/src/constants");

const Message = React.createFactory(require("devtools/client/aboutdebugging-new/src/components/shared/Message"));

describe("Message component", () => {
  it("renders the expected snapshot for INFO level", () => {
    const message = renderer.create(Message({
      children: dom.div({}, "Message content"),
      className: "some-classname",
      level: MESSAGE_LEVEL.INFO,
    }));
    expect(message.toJSON()).toMatchSnapshot();
  });
});
