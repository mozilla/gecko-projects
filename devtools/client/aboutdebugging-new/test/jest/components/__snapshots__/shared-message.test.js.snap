// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders the expected snapshot for INFO level 1`] = `
<aside
  className="message message--level-info js-message some-classname"
>
  <img
    className="message__icon"
    src="chrome://devtools/skin/images/aboutdebugging-information.svg"
  />
  <div
    className="message__body"
  >
    <div>
      Message content
    </div>
  </div>
</aside>
`;
